platform/src/main/java/org/hillview/storage/GenericLogs.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2017 MapleLabs. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview.storage;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.hillview.table.ColumnDescription;
+import org.hillview.table.Schema;
+import org.hillview.table.Table;
+import org.hillview.table.api.ContentsKind;
+import org.hillview.table.api.ITable;
+import org.hillview.storage.FileSetDescription;
+import org.hillview.utils.Utilities;
+
+import java.util.Map;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import io.krakens.grok.api.*;
+
+/**
+ * Reads Generic logs into ITable objects.
+ */
+
+public class GenericLogs {
+    private static Schema schema;
```

Comment:
You cannot use static fields unless they never change; this code will be run on multiple threads and they will stomp on each other's static fields.

Reply:
You don't need these fields to be static anyway.